File: src/main/scala/warmUp/Scala/TImeConversion.scala
/**
  * @author Ram Saran Vuppuluri
  *         Problem statement
  * @see https://www.hackerrank.com/challenges/time-conversion/problem
  */
package warmUp.Scala

import java.util.Scanner

object TImeConversion {
  def main(args: Array[String]): Unit = {
    val scanner = new Scanner(System.in);
    val timeIn = scanner.nextLine().trim
    val timeArray: Array[String] = timeIn.substring(0, 8).split(":")
    if (timeIn.substring(8).toUpperCase == "AM") {
      if (timeArray(0).toInt == 12) {
        println(s"00:${timeArray(1)}:${timeArray(2)}")
      } else {
        println(timeIn.substring(0, 8))
      }
    } else {
      if (timeArray(0).toInt == 12) {
        println(timeIn.substring(0, 8))
      } else {
        println(s"${timeArray(0).toInt + 12}:${timeArray(1)}:${timeArray(2)}")
      }
    }
  }
}

File: src/main/scala/warmUp/Java/TImeConversion.java
/**
 * @author Ram Saran Vuppuluri
 * Problem statement
 * @see https://www.hackerrank.com/challenges/time-conversion/problem
 */
package warmUp.Java;

import java.util.Scanner;

public class TImeConversion {
    public static void main(String... args) {
        Scanner in = new Scanner(System.in);
        String timeIn = in.nextLine().trim();
        if (timeIn.substring(8).toUpperCase().equals("AM")) {
            if ((timeIn.split(":")[0]).equals("12")) {
                System.out.println("00:" + timeIn.substring(3, 8));
            } else {
                System.out.println(timeIn.substring(0, 8));
            }
        } else {
            if ((timeIn.split(":")[0]).equals("12")) {
                System.out.println(timeIn.substring(0, 8));
            } else {
                System.out.println((Integer.parseInt(timeIn.split(":")[0]) + 12) + ":" + timeIn.substring(3, 8));
            }
        }

    }
}
